### PR TITLE
Add error handling for cogs monthly booster

### PR DESCRIPTION
**Pull request by hilmoo**

### CODE REVIEW

File: bot/cogs/booster.py [hunk 1]
```diff
@@ -65,6 +70,13 @@ async def _monthly_booster(self) -> None:
                 file=file,
             )
 
+            if member_error != '':
```
Comment (Iqrar99) [inline bot/cogs/booster.py:73]: Can be simplify to
```suggestion
            if member_error:
```

File: bot/cogs/booster.py [hunk 2]
```diff
@@ -65,6 +70,13 @@ async def _monthly_booster(self) -> None:
...
+                embed = discord.Embed(
+                    title="Error handling user", description=member_error, colour=0xFF0000
```
Comment (Iqrar99) [inline bot/cogs/booster.py:75]: Discord.color.red() would be better for clarity

**Review by Iqrar99 — CHANGES_REQUESTED** [review]

**Comment by hilmoo** [comment]
My latest update:
If an error occurs, the wrapper will return None, which will then be handled in booster.py. Also, now the log is inside wrapper.py.

File: bot/cogs/booster.py [hunk 3]
```diff
@@ -27,10 +28,10 @@ def __init__(self, bot: WarnetBot) -> None:
     async def on_connect(self) -> None:
         self._monthly_booster.start()
 
-    @tasks.loop(time=time(hour=0, minute=0, tzinfo=timezone(timedelta(hours=7))))
+    @tasks.loop(time=time(hour=4, minute=54, tzinfo=timezone(timedelta(hours=7))))
```
Comment (Iqrar99) [inline bot/cogs/booster.py]: why?

File: bot/cogs/booster.py [hunk 4]
```diff
@@ -27,10 +28,10 @@ def __init__(self, bot: WarnetBot) -> None:
...
     async def _monthly_booster(self) -> None:
         date = datetime.now(pytz.timezone('Asia/Jakarta'))
-        if date.day == 1:
+        if date.day == 2:
```
Comment (Iqrar99) [inline bot/cogs/booster.py]: should be 1

File: bot/cogs/booster.py [hunk 5]
```diff
@@ -39,16 +40,19 @@ async def _monthly_booster(self) -> None:
             role = guild.get_role(BOOSTER_ROLE_ID)
             month = date.strftime("%B")
             member_tags = member_ids = member_error = ''
+            succes_count = error_count = 0
```
Comment (Iqrar99) [inline bot/cogs/booster.py:43]:
```suggestion
            success_count = error_count = 0
```

File: bot/cogs/booster.py [hunk 6]
```diff
@@ -39,16 +40,19 @@ async def _monthly_booster(self) -> None:
...
 
             for member in role.members:
-                try:
-                    await TatsuApi().add_score(member.id, BOOSTER_MONTHLY_EXP)
+                result = await TatsuApi().add_score(member.id, BOOSTER_MONTHLY_EXP)
+                if result is not None:
```
Comment (Iqrar99) [inline bot/cogs/booster.py:47]: Can be refactored to
```suggestion
                if result:
```

File: bot/cogs/booster.py [hunk 7]
```diff
@@ -39,16 +40,19 @@ async def _monthly_booster(self) -> None:
...
                     member_ids += f"{member.id} "
-                except Exception as e:
-                    member_error += f"{member.mention} "
-                    logger.error(f'Failed to modify user score. User ID: {member.id}')
-
+                else:
+                    error_count += 1
+                    member_error += f"{member.mention}, "
+            print(len(role.members))
+            print(succes_count)
```
Comment (Iqrar99) [inline bot/cogs/booster.py]: remove debug lines

File: bot/module/tatsu/wrapper.py [hunk 8]
```diff
@@ -124,13 +124,17 @@ async def _modify_score(
...
             logger.error(f"An error occurred: {str(e)}")
+            logger.error(f'Failed to modify user score. User ID: {user_id}')
 
-        score = ds.GuildScoreObject(
-            guild_id=result.get('guild_id'),
-            score=result.get('score'),
-            user_id=result.get('user_id'),
-        )
-        return score
+        if result is not None:
```
Comment (Iqrar99) [inline bot/module/tatsu/wrapper.py:129]:
```suggestion
        if result:
```

**Review by Iqrar99 — CHANGES_REQUESTED** [review]

**Review by Iqrar99 — APPROVED** [review]
LGTM

**Pull request merged**